zerver/forms.py

```diff
@@ -360,6 +361,11 @@ def save(
         if user is not None:
             context["active_account_in_realm"] = True
             context["reset_url"] = generate_password_reset_url(user, token_generator)
+            event = {
+                "type": "soft_reactivate",
+                "user_profile_id": user.id,
+            }
+            queue_json_publish("deferred_work", event)
```

Comment:
I think it might be a bit cleaner to wrap this in a simple helper function `queue_soft_reactivation(user_profile_id: int)` or such - and we can deduplicate the event construction with the analogical block in `soft_reactivate_if_personal_notification`

Reply:
It looks like you forgot to update this commit to call the new function? :)